### ISSUE TITLE
nose is still required to run tests

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#1558]
### Expected behaviour

without a nose installation I can still run all tests

### Actual behaviour

due to our usage of `assert_raises` nose is still needed. Numpy imports it when we call the `assert_raises` function.

**Comment by utkbansal** [comment]
@kain88-de This is being worked on in the individual PRs. Is that okay, or do we need to do this first?

**Comment by kain88-de** [comment]
All of these specific instances can be done in one pr

**Comment by kain88-de** [comment]
This fits into a single PR because the changes are all the same even though you touch a lot of files.

**Comment by kain88-de** [comment]
coordinates/test_trz.py
coordinates/test_reader_api.py
coordinates/test_gro.py
coordinates/base.py
coordinates/test_netcdf.py
coordinates/test_xdr.py
coordinates/test_writer_registration.py
coordinates/test_pdb.py
coordinates/test_dlpoly.py
coordinates/test_dms.py
coordinates/test_trj.py
coordinates/test_dcd.py
coordinates/test_mol2.py
coordinates/test_lammps.py
lib/test_util.py
topology/test_pdb.py
topology/test_gro.py
topology/test_guessers.py
core/test_updating_atomgroup.py
core/test_requires.py
core/test_residuegroup.py
core/test_fragments.py
core/test_universe.py
core/test_groups.py
core/test_group_traj_access.py
core/test_topology.py
core/test_atomselections.py
core/test_topologyobjects.py
core/test_atom.py
analysis/test_density.py
core/test_atomgroup.py
analysis/test_align.py
analysis/test_hydrogenbondautocorrel.py
analysis/test_persistencelength.py
analysis/test_base.py
analysis/test_helanal.py
utils/test_persistence.py
utils/test_distances.py
utils/test_units.py


These are the files that still use assert_raises

**Comment by orbeckst** [comment]
Can you please put `nose` back in? At the moment out-of-the-box testing is broken: If I just install develop from source with the following commands and run the tests
```
pip install numpy cython
pip install package/
pip install testsuite/
py.test testsuite/MDAnalysisTests
```

then tests fail completely:

```
(mda_py2) oliver@spudda:~/Library/python/mdanalysis$ py.test testsuite/MDAnalysisTests
=========================================== test session starts ===========================================
platform linux2 -- Python 2.7.12, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /nfs/homes2/oliver/Library/python/mdanalysis/testsuite, inifile:
plugins: hypothesis-3.14.0
collected 3894 items / 17 errors

================================================= ERRORS ==================================================
_________________________ ERROR collecting MDAnalysisTests/analysis/test_align.py _________________________
ImportError while importing test module '/nfs/homes2/oliver/Library/python/mdanalysis/testsuite/MDAnalysisTests/analysis/test_align.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
testsuite/MDAnalysisTests/analysis/test_align.py:296: in <module>
    class TestAlignmentProcessing(TestCase):
testsuite/MDAnalysisTests/analysis/test_align.py:316: in TestAlignmentProcessing
    msg="Test skipped because clustalw2 executable not found")
../../../.virtualenvs/mda_py2/local/lib/python2.7/site-packages/numpy/testing/decorators.py:123: in skip_decorator
    import nose
E   ImportError: No module named nose
________________________ ERROR collecting MDAnalysisTests/analysis/test_encore.py _________________________
ImportError while importing test module '/nfs/homes2/oliver/Library/python/mdanalysis/testsuite/MDAnalysisTests/analysis/test_encore.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
testsuite/MDAnalysisTests/analysis/test_encore.py:45: in <module>
    class TestEncore(TestCase):
testsuite/MDAnalysisTests/analysis/test_encore.py:369: in TestEncore
    @dec.slow
../../../.virtualenvs/mda_py2/local/lib/python2.7/site-packages/numpy/testing/decorators.py:123: in skip_decorator
    import nose
E   ImportError: No module named nose

...
```

*This is not good.*

Please make sure that you *also test locally* and not only rely on Travis where we set up our own environment. It is not good if I want to tell a user "follow our docs" just to note that this is broken.

Furthermore, is parallel testing disabled? When I do the above I don't get any options to run tests in parallel.

**Comment by kain88-de** [comment]
@utkbansal please fix this!

**Comment by utkbansal** [comment]
Yeah, I'm working on it.

**Comment by kain88-de** [comment]
Nose is no longer needed.
